core/include/seqan/align/evaluate_alignment.h

```diff
@@ -252,6 +252,241 @@ TScoreVal computeAlignmentStats(Align<TGaps, TAlignSpec> const & align,
     return computeAlignmentStats(stats, align, scoringScheme);
 }
 
+// ----------------------------------------------------------------------------
+// Function alignmentIdentityString()
+// ----------------------------------------------------------------------------
+
+/*!
+ * @fn Align#alignmentIdentityString()
+ * @headerfile <seqan/align.h>
+ * @brief Returns a CharString of the similarity of positions in an alignment
+ *
+ * @signature CharString alignmentIdentityString();
+ *
+ * @param align The @link Align @endlink object to score for similarity.
+ * @param match The character to use for similarity at match positions
+ * @param mismatch The character to use for similarity at mismatch positions
+ * @param gap The character to use for similarity at gap positions
+ *
+ * @signature CharString A string of characters representing alignment position similarity
+ *
+ * @section Examples
+ *
+ * @include demos/align/compute_match_string.cpp
+ *
+ * The output is as follows:
+ *
+ * @include demos/align/compute_match_string.cpp.stdout
+ */
+
+
+template <typename TGaps>
+String<char> alignmentIdentityString(TGaps & queryRow,
```

Comment:
The function can be simplified using the std::transform interface.
Just add a function as follows:
``` cpp
template <typename TTarget, typename TSourceH, typename TSpecH, typename TSourceV, typename TSpecV, typename TBinaryOperator>
inline void transform(TTarget & target, 
                      Gaps<TSourceH, TGapsH> const & gapsH,
                      Gaps<TSourceV, TGapsV> const & gapsV,
                      TBinaryOperator const & binOperator)
{
    SEQAN_ASSERT_EQ(length(gapsH), length(gapsV));

    resize(target, length(gapsH), Exact());
    std::transform(begin(gapsH, Standard()), end(gapsH, Standard()), begin(gapsV, Standard()), begin(target, Standard()), binOperator);
}
```

At the moment we don't have the parallel version of transform implemented but we could easily switch later to use to the generic version (not before the release :)), which can then also be parallelised.

Then you simply create a functor that does the steps from line 310 to 326, except that your function-operator just returns the char that should be written to the target container. I added comments to the lines below.
The functor is then initialised with the chars for the different events.